packages/drivers/routerlicious-driver/src/test/mapWithExpiration.spec.ts

```diff
@@ -283,24 +283,24 @@ describe("MapWithExpiration", () => {
 			},
 		);
 
-		// In ESM the the 'this' value is 'undefined', but in CJS it is '{__esModule: true}'.
-		// Therefore, we exempt the test value 'undefined' from the below assertion.
-		if (this !== undefined) {
-			testForEachCases("Arrow functions don't pick up thisArg", (maps, thisArgs) => {
-				for (const thisArg of thisArgs) {
-					for (const map of maps) {
-						map.set(1, "one");
-						map.forEach(() => {
-							assert.notEqual(
-								this,
-								thisArg,
-								"Expected 'this' to be unchanged for arrow fn",
-							);
-						}, thisArg);
-					}
+		testForEachCases("Arrow functions don't pick up thisArg", (maps, thisArgs) => {
+			for (const thisArg of thisArgs) {
+				for (const map of maps) {
+					map.set(1, "one");
+
+					// eslint-disable-next-line @typescript-eslint/no-this-alias
+					const thisOutsideForEach = this;
```

Comment:
To avoid the different cases for global `this`, maybe we should put the `map.forEach` call inside a class function.

Reply:
Like [this](https://www.typescriptlang.org/play?#code/MYewdgzgLgBANiA5jAvDUkRwKYDoGIDcAsAFBkBmArmMFAJbgxTbQAUAlDAN5kz8D+AeiHNWsCNihUADjGhUKFPoIEZoMALYBDOWjDYA7jACyutgG0LARgA0MAEThsDgLquOJcqVUC2B4zZgOG0ICB5mEABlKAAnejBETgjYqSpYsEcKEBAHQhgAXxhYmgAVcQBhUOxk3h9fBuFRKOxsewADKAALegh2mF7mLux0ELCByChtWjaYMBAJEE0RxAQAI204IYTEXBVG3wI2doASbm7egvbPfYPBW7v+dVgLiAB5KigIegATEbRXl5HqpntsIABBWKIAD69EQ81SP1QETWIS6AC5HIYuiB0NpUhAHIUgcCBDoZLhsrEAKLaYBdNjJFAAPngSDYr1QKABPXen2+fw49lekJhcIR2B+N3qvgKBQ4HFwJTA5WgVUknC8BTIZBY7E8QA)